src/main/scala/Engine/Architecture/Worker/Processor.scala

```diff
@@ -210,6 +214,24 @@ class Processor(val dataProcessor: TupleProcessor,val tag:WorkerTag) extends Wor
       AdvancedMessageSending.nonBlockingAskWithRetry(context.parent,ReportWorkerPartialCompleted(tag,from),10,0)
   }
 
+  final def allowOperatorLogicUpdate:Receive = {
+    case ModifyLogic(newLogic) =>
+      // newLogic is something like {"operatorID":"Filter","operatorType":"Filter","targetField":2,"filterType":"Greater","threshold":"1991-01-01"}
+      val json: JsValue = Json.parse(newLogic)
+      val operatorType = json("operatorID").as[String]
+      json("operatorType").as[String] match{
+        case "KeywordMatcher" =>
+          var dataProcessor: KeywordSearchTupleProcessor = dataProcessor.asInstanceOf[KeywordSearchTupleProcessor]
+          dataProcessor.setPredicate(json("attributeName").as[Int],json("keyword").as[String])
+        case "Filter" =>
+          var dataProcessor: FilterSpecializedTupleProcessor = dataProcessor.asInstanceOf[FilterSpecializedTupleProcessor]
+          dataProcessor.filterType = FilterType.getType[DateTime](json("filterType").as[String])
```

Comment:
This line is giving me compile error. @shengquan-ni can you help me in setting filterType correctly?

Reply:
`filterType` is not used in `FilterSpecializedTupleProcessor` class. I have applied a fix.